packages/core/src/browser/quick-input/quick-command-service.ts

```diff
@@ -31,7 +33,8 @@ export const CLEAR_COMMAND_HISTORY: Command = {
 };
 
 @injectable()
-export class QuickCommandService extends QuickInputService implements QuickAccessContribution {
+export class QuickCommandService implements QuickAccessContribution, QuickAccessProvider {
+    static PREFIX = '>';
```

Comment:
Changes in this file do not seem to work correctly.
The list of recently used commands are not preserved when re-triggering the quick-command.

The list of recently used commands only appear upon reloading the application, performing `clear command history` also does not update the list until the app is reloaded.

Reply:
I can't reproduce this issue in the browser (Firefox) environment:

https://user-images.githubusercontent.com/4377073/127165072-d4797bde-4b8d-4ef4-8b97-af0caf3a0c88.mp4

Reply:
@msujew it does not work for me in the browser (chrome + firefox), and electron. The commands are not present in the recently used list until the app is reloaded.

Reply:
This seems to work just fine for me, as well (both the "recently used" commands as well as the "clear history" part. Could you maybe give more detailed steps to reproduce?

Reply:
@tsmaeder I'm using Ubuntu. it's likely its my environment since Mark, Alvaro and yourself all don't experience the issue while I do even when building multiple times. 

If interested, to reproduce I do the following to have a clean environment:
- perform `rm -rf ~/.theia` && `rm -rf ~/.config/Theia\ Electron\ Example` (to remove any config or cache)
- start the electron app
- execute `reset workbench layout`
- execute `clear command history`
- close the app through the window `x`
- restart the app
- perform the tests, see if executing commands adds them to the recently used list and so on

Note that prior to the changes the recently used list did work for me.

Reply:
Just to mention, I built on a completely different machine (macos) and the problem persists for me.

Reply:
I don't know what to do here: this works perfectly for me and I've inspected the code and my changes just don't look like they could cause the behaviour you mention. Could you give the exact steps you're doing, including

1. The name of the commands you're invoking
2. How you're invoking them (menu, etc.)
3. Open editors, views, etc?

/Thomas

Reply:
@tsmaeder I'm not sure either, I've rebuilt the pull-request a couple of times and on different machines but I can retry:
1. the commands are mainly random, only to see if they are populated in the recent list and they only are after reload.
2. invoking the commands through the `command palette` itself.
3. the view is the default since I performed `reset workbench layout` to have a clean reproducible state.

Reply:
And by `command palette`, you mean the "quick access" invoked by hitting F1?

Reply:
Last resort: make a video 🤷 ?

Reply:
@tsmaeder yes correct, the quick-command-service through <kbd>F1</kbd>. 
I managed to do a fresh clone in a new directory, and rebuild your changes and they seem to be working fine now, I'm really sorry about that.

Reply:
Javascript build systems for teh win! Personally, I like `git clean -xfd` ;-)

Reply:
I always do it as part of my workflow so not really sure what happened (the fact it even happened on other machines as well). Anyways please proceed :)